clippy_lints/src/vec.rs

```diff
@@ -185,6 +185,11 @@ impl UselessVec {
         let snippet = match *vec_args {
             higher::VecArgs::Repeat(elem, len) => {
                 if let Some(Constant::Int(len_constant)) = constant(cx, cx.typeck_results(), len) {
+                    // vec![ty; N] works when ty is Clone, [ty; N] requires it to be Copy also
+                    if !is_copy(cx, cx.typeck_results().expr_ty(elem)) {
```

Comment:
This is also checked up top, but only in one case:

https://github.com/rust-lang/rust-clippy/blob/9907b90b1e014fb910c082612b2627dd375704fe/clippy_lints/src/vec.rs#L82
Can this be removed if it's checked here instead?

Reply:
Indeed, it looks like checking it before emitting the lint is sufficient. It even allows removal of the `vec_type()` function since we check the type of the copied element instead.